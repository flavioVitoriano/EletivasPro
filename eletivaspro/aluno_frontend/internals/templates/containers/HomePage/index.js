/**
 *
 * HomePage
 *
 * This is the first thing users see of our App, at the '/' route
 *
 */

import React from 'react';
import { FormattedMessage } from 'react-intl';

const HomePage = () => (
  <div>
    <FormattedMessage id="containers.homePage.header" />
  </div>
);

export default HomePage;
