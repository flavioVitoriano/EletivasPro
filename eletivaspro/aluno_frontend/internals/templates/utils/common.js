import produce from 'immer';
import assign from 'lodash/assign';
import isString from 'lodash/isString';
import keys from 'lodash/keys';
import reduce from 'lodash/reduce';

export const flattenMessages = (nestedMessages, prefix = '') =>
  reduce(
    keys(nestedMessages),
    (messages, key) => {
      const value = nestedMessages[key];
      const prefixedKey = prefix ? `${prefix}.${key}` : key;

      if (isString(value)) {
        return produce(messages, draft => {
          draft[prefixedKey] = value;
        });
      }
      return assign(messages, flattenMessages(value, prefixedKey));
    },
    {},
  );
