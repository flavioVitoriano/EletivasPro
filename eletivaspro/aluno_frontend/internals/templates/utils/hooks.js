import { useCallback } from 'react';
import {
  shallowEqual,
  useDispatch,
  useSelector as useReduxSelector,
} from 'react-redux';

export const useAction = action => {
  const dispatch = useDispatch();
  return useCallback(prop => dispatch(action(prop)), [action, dispatch]);
};

export const useSelector = selector => useReduxSelector(selector, shallowEqual);
