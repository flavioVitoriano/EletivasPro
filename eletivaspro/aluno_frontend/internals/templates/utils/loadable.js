/* eslint-disable react/jsx-props-no-spreading */
import React, { lazy, Suspense } from 'react';

const loadable = (importFunc, { fallback = null } = { fallback: null }) => {
  const LazyComponent = lazy(importFunc);

  return props => (
    <Suspense fallback={fallback}>
      <LazyComponent {...props} />
    </Suspense>
  );
};

export default loadable;
