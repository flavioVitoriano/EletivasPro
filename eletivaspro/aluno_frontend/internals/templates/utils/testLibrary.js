import { render } from '@testing-library/react';
import LanguageProvider from 'containers/LanguageProvider';
import PropTypes from 'prop-types';
import React from 'react';
import { Provider } from 'react-redux';
import { store } from 'store';

const AllTheProviders = ({ children }) => (
  <Provider store={store}>
    <LanguageProvider>{children}</LanguageProvider>
  </Provider>
);

AllTheProviders.propTypes = {
  children: PropTypes.node,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

const HooksWrapper = ({ children, hooks }) => children(hooks());

export * from '@testing-library/react';

// override render method
export { HooksWrapper, customRender as render };
