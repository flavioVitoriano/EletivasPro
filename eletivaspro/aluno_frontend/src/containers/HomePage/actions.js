/**
 *
 * HomePage actions
 *
 */

import {
  ADD_COUNTER,
  GET_USER_FAILURE,
  GET_USER_REQUEST,
  GET_USER_SUCCESS,
  SUBTRACT_COUNTER,
} from './constants';

export const addCounter = () => ({
  type: ADD_COUNTER,
});

export const subtractCounter = () => ({
  type: SUBTRACT_COUNTER,
});

export const getUser = id => ({
  type: GET_USER_REQUEST,
  id,
});

export const getUserSuccess = user => ({
  type: GET_USER_SUCCESS,
  user,
});

export const getUserError = error => ({
  type: GET_USER_FAILURE,
  error,
});
