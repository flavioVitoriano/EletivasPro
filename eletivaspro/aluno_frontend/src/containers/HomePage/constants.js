/**
 *
 * HomePage constants
 *
 */

export const ADD_COUNTER = 'HomePage/ADD_COUNTER';
export const SUBTRACT_COUNTER = 'HomePage/SUBTRACT_COUNTER';
export const GET_USER_REQUEST = 'HomePage/GET_USER';
export const GET_USER_SUCCESS = 'HomePage/GET_USER_SUCCESS';
export const GET_USER_FAILURE = 'HomePage/GET_USER_FAILURE';
