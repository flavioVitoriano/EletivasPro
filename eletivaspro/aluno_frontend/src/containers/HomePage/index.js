/**
 *
 * HomePage
 *
 */

import { Button, Select, Title } from 'components';
import { logoutUser } from 'containers/App/actions';
import { changeLocale } from 'containers/LanguageProvider/actions';
import { makeSelectLocale } from 'containers/LanguageProvider/selectors';
import React, { useEffect } from 'react';
import { useIntl } from 'react-intl';
import { useAction, useSelector } from 'utils/hooks';
import { locales, messages } from 'utils/locale';

import { addCounter, getUser, subtractCounter } from './actions';
import { makeSelectCounter } from './selectors';

const HomePage = () => {
  const { formatMessage } = useIntl();
  const counter = useSelector(makeSelectCounter());
  const locale = useSelector(makeSelectLocale());
  const addCounterCallback = useAction(addCounter);
  const subtractCounterCallback = useAction(subtractCounter);
  const changeLocaleCallback = useAction(changeLocale);
  const getPlaceholderUserCallback = useAction(getUser);
  const logoutUserCallback = useAction(logoutUser);

  useEffect(() => {
    getPlaceholderUserCallback(counter);
  }, [getPlaceholderUserCallback, counter]);

  return (
    <>
      <Title text={formatMessage({ id: 'containers.homePage.header' })} />
      <Button onClick={addCounterCallback}>+</Button>
      {counter}
      <Button onClick={subtractCounterCallback}>-</Button>
      <Select
        value={locale}
        options={locales}
        messages={messages}
        onSelect={changeLocaleCallback}
      />
      <Button onClick={logoutUserCallback}>
        {formatMessage({ id: 'containers.homePage.logout' })}
      </Button>
    </>
  );
};

export default HomePage;
