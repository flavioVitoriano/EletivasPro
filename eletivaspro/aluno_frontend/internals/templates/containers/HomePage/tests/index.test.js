/**
 *
 * Tests for HomePage
 *
 */

import React from 'react';
import { render } from 'utils/testLibrary';

import HomePage from '../index';

describe('<HomePage />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const dispatch = jest.fn();
    render(<HomePage dispatch={dispatch} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Expect to have additional unit tests specified', () => {
    // TODO: Implement unit tests
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<HomePage />);
    expect(firstChild).toMatchSnapshot();
  });
});
