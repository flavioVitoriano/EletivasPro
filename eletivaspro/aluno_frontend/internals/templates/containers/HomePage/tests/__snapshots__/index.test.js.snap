// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> Should render and match the snapshot 1`] = `
<div>
  This is the HomePage container!
</div>
`;
